pyannotate_runtime/collect_types.py

```diff
@@ -642,6 +643,13 @@ def _make_sampling_sequence(n):
 call_pending = set()  # type: Set[int]
 
 
+@contextmanager
+def collect():
+    resume()
+    yield
+    pause()
```

Comment:
It would be good to add a try/finally here so pause() is called even if the wrapped function raises an exception.